### PR TITLE
Normalize line endings and improve linting

### PR DESCRIPTION
**Pull request by MeekLogic**
This PR addresses line endings and further enforces code styling through ESLint.

**Line endings** will be handled automatically by git via the `.gitattributes` config.

**Linting additions (w/ recommend rulesets)**

- [`eslint-plugin-ava`](https://github.com/avajs/eslint-plugin-ava)
- [`eslint-plugin-node`](https://github.com/mysticatea/eslint-plugin-node)
- [`eslint-plugin-import`](https://github.com/benmosher/eslint-plugin-import)
- [`eslint-config-google`](https://github.com/google/eslint-config-google)

If you prefer a different ESLint ruleset (or prefer to do it manually) then we can do that instead. JavaScript does not have a "standard" code styling.

Here are some alternatives:

- [JavaScript Standard Style](https://standardjs.com/)
- [Prettier](https://prettier.io/)
- [Airbnb](https://github.com/airbnb/javascript)

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  "parserOptions": {
+    "ecmaVersion": 2020,
+    "sourceType": "module"
+  },
+  "rules": {
+    "max-len": "warn",
+    "no-invalid-this": "warn",
+    "camelcase": "warn",
+    "require-jsdoc": "warn"
```
Comment (forzagreen) [inline .eslintrc.json:30]: I see that this rule was deprecated: https://eslint.org/docs/rules/require-jsdoc

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Automatically handle line endings
+* text=auto
+
+# Force line endings for these files
+*.js  text eol=lf
+*.cjs text eol=lf
+*.mjs text eol=lf
+
+# Don't add these files to git archives
+CHANGELOG.md export-ignore
```
Comment (forzagreen) [inline .gitattributes:10]: I would keep the `CHANGELOG.md` file. It's a small file with very useful information.

File: .eslintrc.json (hunk 1 above)
Comment (MeekLogic) [reply .eslintrc.json:30]: This rule is still included within ESLint and was activated from `eslint-config-google`.

This corresponding line reduces the errors into warnings. If you do not want to show warnings for missing JSDoc then we can turn `require-jsdoc` to `off`.

File: .gitattributes (hunk 2 above)
Comment (MeekLogic) [reply .gitattributes:10]: No problem. I'll update the `.gitattributes`.

File: .eslintrc.json (hunk 1 above)
Comment (MeekLogic) [reply .eslintrc.json:30]: I see that ESLint [recommends](https://eslint.org/blog/2018/11/jsdoc-end-of-life) using the community supports [`eslint-plugin-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc)

Comment (forzagreen) [reply .eslintrc.json:30]: All right. Let's keep it then.

**Pull request merged**